python/gink/impl/key_set.py

```diff
@@ -0,0 +1,222 @@
+        found = self._database.get_store().get_entry_by_key(self.get_muid(), key=key, as_of=as_of)
+
+        return False if found is None or found.builder.deletion else True
+
+    def discard(self, key: UserKey, bundler: Optional[Bundler]=None, comment: Optional[str]=None):
+        """ Deletes a specified entry from the key set """
+        return self._add_entry(key=key, value=deletion, bundler=bundler, comment=comment)
+    
+    def remove(self, key: UserKey, bundler: Optional[Bundler]=None, comment: Optional[str]=None):
+        """ Deletes a specified entry from the key set, but returns KeyError if not found """
+        as_of = self._database.get_now()
+        found = self._database.get_store().get_entry_by_key(self.get_muid(), key=key, as_of=as_of)
+        if found is None or found.builder.deletion:  # type: ignore
+            raise KeyError("Key does not exist")
+        else:
+            return self._add_entry(key=key, value=deletion, bundler=bundler, comment=comment)
+    
+    def pop(self, key: UserKey, bundler: Optional[Bundler]=None, comment: Optional[str]=None, default=None):
+        """ If key exists in the mapping, returns the corresponding value and removes it.
+
+            Otherwise returns default.  In the case that the key is found and removed,
+            then the change is added to the bundler (or committed immedately with comment
+            if no bundler is specified.)
+        """
+        as_of = self._database.get_now()
+        found = self._database.get_store().get_entry_by_key(self.get_muid(), key=key, as_of=as_of)
+        if found is None or found.builder.deletion:  # type: ignore
+            return default
+        self._add_entry(key=key, value=deletion, bundler=bundler, comment=comment)
+        return decode_key(found.builder)
+    
+    def issuperset(self, subset: Iterable, *, as_of: GenericTimestamp=None) -> bool:
+        """ Returns a Boolean stating whether the key set contains the specified set or list of keys """
+        for element in subset:
+            if not self.contains(element):
+                return False
+        return True
+    
+    def issubset(self, superset: Iterable, *, as_of: GenericTimestamp=None) -> bool:
+        """ Returns a Boolean stating whether the key set is a subset of the specified set/list/tuple """
+        as_of = self._database.resolve_timestamp(as_of)
+        iterable = self._database.get_store().get_keyed_entries(
+            container=self.get_muid(), behavior=self.BEHAVIOR, as_of=as_of)
+        for entry_pair in iterable:
+            if entry_pair.builder.deletion:
+                continue
+            if decode_key(entry_pair.builder) not in superset:
+                return False
+        return True
+    
+    def isdisjoint(self, s: Iterable, *, as_of: GenericTimestamp=None) -> bool:
+        """ Returns a boolean stating whether the key set contents completely overlap with the specified set/list/tuple 
```

Comment:
I think you want to remove the `completely` from the doc-string.
Also, when you use the `Iterable` type annotation, you should make it specific, e.g. `Iterable[UserKey]`